From 93e4b73f4f8e83955b6acdfd65876af9de49ea82 Mon Sep 17 00:00:00 2001
From: Izak Burger <isburger@gmail.com>
Date: Thu, 18 Mar 2021 15:05:22 +0200
Subject: [PATCH 1/4] MFD: Avoid race using Multi as battery measurement

The general rule is that if /Dc/0/Voltage is invalid, then
a measurement cannot be used. This rule may affect the Multi
if the vebus service don't have /Dc/0/Voltage immediately after
becoming available on dbus. This may cause a delay in the multi
showing up as a battery measurement for MFD integration, or it
may not show up at all. But since the Multi cannot operate without
a battery anyway, this general rule is of no use and we can simply
always use the Multi as a battery measurement.

https://github.com/victronenergy/venus/issues/802
---
 delegates/batterydata.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/delegates/batterydata.py b/delegates/batterydata.py
index 063c0ad..9bdb3a8 100644
--- a/delegates/batterydata.py
+++ b/delegates/batterydata.py
@@ -174,6 +174,11 @@ class MultiTracker(BatteryTracker):
 		super(MultiTracker, self).__init__(service, instance, monitor)
 		self._dbusservice = dbusservice
 
+	@property
+	def valid(self):
+		# The Multi is always a valid battery measurement
+		return True
+
 class BatteryData(SystemCalcDelegate):
 	def __init__(self):
 		SystemCalcDelegate.__init__(self)
-- 
2.7.4

